core/lib/spree/testing_support/factories/shipping_method_factory.rb

```diff
@@ -10,6 +10,12 @@
       end
     end
 
+    after(:create) do |shipping_method, evaluator|
+      if shipping_method.stock_locations.empty?
+        shipping_method.stock_locations << (Spree::StockLocation.all.to_a)
```

Comment:
@jhawthorn This is quite an issue, and I was just trying to get my specs to go green. 

However, since I have looked around and it's a difficult one to find. This is because with this PR we now have a dependency between: `shipping_method` and `stock_location`. In most factories these get created separately and for the code to work, we need to link them.

So now we have two solutions:

1) If we don't do this here, we'd have to replicate it through most factories, and my gut feeling is that this will be quite difficult in some particular cases (ie: factories which call other factories...)

2) Do this here and try and find a way to clean it up.

---